### PR TITLE
get first repo with todos

### PR DESCRIPTION
**Pull request by unclejustin**
Closes https://github.com/sourcegraph/sourcegraph/issues/30745

Updates the dynamic example to check for an actual repo with TODOs. If no repo is found containing "TODO" then the first repo is used.

Note: We are assuming that all customers will have at least 1 repo. In the unlikely event that no repo is found then an empty string will be returned.

PixelSnap 2022-02-15 at 22 49 50

### CODE REVIEW

File: client/web/src/enterprise/insights/core/backend/gql-api/code-insights-gql-backend.ts [hunk 1]
```diff
@@ -453,4 +456,28 @@ export class CodeInsightsGqlBackend implements CodeInsightsBackend {
             )
         )
     }
+
+    public getFirstExampleRepository = (): Observable<string> => {
+        const firstRepository = (): Observable<string> =>
+            fromObservableQuery(
+                this.apolloClient.watchQuery<ExampleFirstRepositoryResult>({
+                    query: GET_EXAMPLE_FIRST_REPOSITORY_GQL,
```
[the remark this answers is not in the document]
Comment (coury-clark) [reply client/web/src/enterprise/insights/core/backend/gql-api/code-insights-gql-backend.ts:464]: We can try it on k8s which is a fairly large installation (though lacking any large monorepos). This has the small risk of being *extremely* broken if the search lands on a monorepo.

[the remark this answers is not in the document]
Comment (unclejustin) [reply client/web/src/enterprise/insights/core/backend/gql-api/code-insights-gql-backend.ts:464]: As per @limitedmage this query runs streaming search in the backend. @coury-clark  Am I missing something? This should be just as fast as using streaming search directly.

[the remark this answers is not in the document]
Comment (coury-clark) [reply client/web/src/enterprise/insights/core/backend/gql-api/code-insights-gql-backend.ts:464]: The search backend streams results into the GraphQL (though this isn't the stream API), that's true. Maybe this is a non-issue then, if we include a `limit` clause maybe the search will cancel appropriately

[the remark this answers is not in the document]
Comment (unclejustin) [reply client/web/src/enterprise/insights/core/backend/gql-api/code-insights-gql-backend.ts:464]: This sounds reasonable, in fact, I just remembered @limitedmage told me to do that very thing! 😅

File: client/web/src/enterprise/insights/pages/getting-started/components/dynamic-code-insight-example/DynamicCodeInsightExample.tsx [hunk 2]
```diff
@@ -65,6 +68,12 @@ export const DynamicCodeInsightExample: React.FunctionComponent<DynamicCodeInsig
     })
 
     const hasValidLivePreview = repositories.meta.validState === 'VALID' && query.meta.validState === 'VALID'
+    const repository = useObservable(useMemo(() => getFirstExampleRepository(), [getFirstExampleRepository]))
+
+    // This is to prevent resetting the name in an endless loop
+    if (repository && repositories.input.value !== repository) {
```
[the remark this answers is not in the document]
Comment (vovakulikov) [reply client/web/src/enterprise/insights/pages/getting-started/components/dynamic-code-insight-example/DynamicCodeInsightExample.tsx:74]: I created a follow up issue for this https://github.com/sourcegraph/sourcegraph/issues/31340

**Pull request merged**